service/src/main/java/com/ge/predix/acs/privilege/management/ResourcePrivilegeManagementController.java

```diff
@@ -42,35 +37,61 @@
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
-import com.ge.predix.acs.commons.web.BaseRestApi;
-import com.ge.predix.acs.commons.web.RestApiException;
-import com.ge.predix.acs.commons.web.UriTemplateUtils;
-import com.ge.predix.acs.rest.BaseResource;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 
-import io.swagger.annotations.ApiOperation;
-import io.swagger.annotations.ApiResponse;
-import io.swagger.annotations.ApiResponses;
+import static com.ge.predix.acs.commons.web.AcsApiUriTemplates.MANAGED_RESOURCES_URL;
+import static com.ge.predix.acs.commons.web.AcsApiUriTemplates.MANAGED_RESOURCE_URL;
+import static com.ge.predix.acs.commons.web.AcsApiUriTemplates.V1;
+import static com.ge.predix.acs.commons.web.ResponseEntityBuilder.created;
+import static com.ge.predix.acs.commons.web.ResponseEntityBuilder.noContent;
+import static com.ge.predix.acs.commons.web.ResponseEntityBuilder.notFound;
+import static com.ge.predix.acs.commons.web.ResponseEntityBuilder.ok;
+import static org.springframework.web.bind.annotation.RequestMethod.DELETE;
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
+import static org.springframework.web.bind.annotation.RequestMethod.POST;
+import static org.springframework.web.bind.annotation.RequestMethod.PUT;
 
 /**
  *
  * @author 212319607
  */
 @RestController
 public class ResourcePrivilegeManagementController extends BaseRestApi {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourcePrivilegeManagementController.class);
+
     @Autowired
     private PrivilegeManagementService service;
 
+    private void failIfParentsSpecified(final List<BaseResource> resources) {
+        for (BaseResource resource : resources) {
+            if (!CollectionUtils.isEmpty(resource.getParents())
+                && !Arrays.asList(this.getEnvironment().getActiveProfiles()).contains("titan")) {
```

Comment:
The check appears to be independent of resources we are iterating in for loop, can we take this if statement !Arrays.asList(this.getEnvironment().getActiveProfiles()).contains("titan") out of the for loop